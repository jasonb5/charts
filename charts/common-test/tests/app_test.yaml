suite: test app
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should create extra objects
    set:
      extraObjects:
        - apiVersion: v1
          data:
            CONFIG: /config/config.txt
            PORT: 80
          kind: ConfigMap
          metadata:
            name: test
        - |
          apiVersion: v1
          data:
            CONFIG: /config/config.txt
            PORT: {{ .Values.service.ports.default.port }}
          kind: ConfigMap
          metadata:
          {{- include "common.metadata" . | nindent 2 }}

      service:
        ports:
          default:
            port: 80
    asserts:
      - hasDocuments:
          count: 3
  - it: should create extra workload
    set:
      workload: deployment

      extraWorkloads:
        db:
          workload: deployment

          image:
            repository: mariadb
            tag: latest
    asserts:
      - matchSnapshot: {}
        documentIndex: 1
      - hasDocuments:
          count: 3
  - it: should pass
    asserts:
      - hasDocuments:
          count: 1
