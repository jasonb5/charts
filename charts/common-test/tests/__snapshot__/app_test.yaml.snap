should create extra workload:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        common/workload: db
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test-db
      namespace: NAMESPACE
    spec:
      replicas: null
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: common-test
          common/workload: db
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: common-test
            common/workload: db
        spec:
          containers:
            - image: mariadb:latest
              name: common-test
